ctsit/nacculator
# Add Python DB Logger

## What does this change do?
This change adds Python DB Logger module to Nacculator, to log directly to our logging database.

## Why was this change made?
This change fixes #134 

## Verification
1. First, copy the example env file:
`cp db_logger.env.example .env`
2. Then, start a MySQL instance. This will create the database and the relevant tables and set up Adminer for easy database visualization.
`docker compose -f .containers/mysql/docker-compose.yml up -d`
3. Run Nacculator with the flags and configurations of your choice.
4. Once Nacculator has run successfully, you should find a `report.xls` file in Nacculator's root directory.
5. Open the XLS file and verify that the expected logs (Success, Skip, Failure) are logged.
6. Next, go to Adminer (http://localhost:8080). DB logger database is already configured. Login with the database credentials found in the env file.
7. Verify the table `detail_log` has the logs.

Both the database and XLS will have the same logs you would expect from the stdout and stderr file outputs.

No other configurations are needed, DB Logger has already been set up to start logging automatically when Nacculator script starts running.

## Affirmations
